__tests__/app.test.js

```diff
@@ -0,0 +1,47 @@
+const app = require('../app');
+const request = require('supertest');
+const db = require('../db/connection.js');
+const testData = require('../db/data/test-data');
+const seed = require('../db/seeds/seed');
+
+beforeEach(() => {
+  return seed(testData);
+});
+
+afterAll(() => {
+  db.end();
+});
+
+describe('GET/api/categories', () => {
+  test('GET /api/categories 200 code responds with an array', () => {
+    return request(app)
+      .get('/api/categories')
+      .expect(200)
+      .then(({ body: { categories } }) => {
+        expect(categories).toBeInstanceOf(Array);
+      });
+  });
+  test('GET /api/categories 200 code responds with an array of category object, each should have the properties slug and description', () => {
+    return request(app)
+      .get('/api/categories')
+      .expect(200)
+      .then(({ body: { categories } }) => {
+        categories.forEach((category) => {
+          expect(category).toEqual({
+            slug: expect.any(String),
+            description: expect.any(String),
+          });
```

Comment:
Id like a test for the length of the array. A forEach will always pass on an empty array so we have to check that it has the correct length first

Reply:
I have added the requested amendment, and committed the change... do I need to generate a new pull request - or is this sufficient?